### PR TITLE
Migrate start command to cobra

### PR DESCRIPTION
**Pull request by WeiZhang555**
Part of #23211 

Signed-off-by: Zhang Wei zhangwei555@huawei.com

ping @vdemeester, I'd like to help with the migration, hope you haven't start doing the `start` command, 

/cc @dnephin

### CODE REVIEW

**Comment by WeiZhang555** [comment]
@vdemeester May I also take `attach`, `kill`, `restart` and `rm`(randomly picked)?
 I noticed that only you are working on this, if you already start working on one of them, I'll leave it to you :smile:

**Comment by vdemeester** [comment]
@WeiZhang555 yay cool !! For now @dnephin and I are working on this so you're help is more than welcome (and should help to make all commands into 1.12 :angel:)

I noted you're name (and the PR) in https://github.com/docker/docker/issues/23211 :wink:

File: api/client/utils.go [hunk 1]
```diff
@@ -203,3 +203,34 @@ func (cli *DockerCli) retrieveAuthConfigs() map[string]types.AuthConfig {
 	acs, _ := getAllCredentials(cli.configFile)
 	return acs
 }
+
+// ForwardAllSignals forwards signals to the contianer
+// TODO: this can be unexported again once all container commands are under
```
Comment (vdemeester) [inline api/client/utils.go]: :+1:

**Comment by dnephin** [comment]
LGTM

**Comment by thaJeztah** [comment]
Oh, sorry @WeiZhang555, this needs a rebase now

**Comment by WeiZhang555** [comment]
@thaJeztah Rebased. :+1:

**Comment by vdemeester** [comment]
LGTM 🐮

**Pull request merged**